### PR TITLE
Support Bazel 0.5.3

### PR DESCRIPTION
**Pull request by jml**
It's out, and it has improved cross-compilation support.

Getting this to work was a bit of a saga. See #530 for the gruesome details. In terms of meaningful changes:

- constrain zipkin-go-opentracing to 0.3.x versions, to workaround https://github.com/bazelbuild/rules_go/issues/423
- explicitly depend on `*.pb.go` files in the relevant make targets, so we get proper gazelle files, and so the build passes
- (probably?) drop support for earlier bazels. Maybe we still work with them, but I don't think we want to put the effort in to make sure that's the case, so just document that people need something recent.

Haven't actually tried cross-compiling.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -18,3 +18,10 @@
 [[override]]
   name = "github.com/weaveworks/mesh"
   revision = "5015f896ab62d3e9fe757456c757521ce0c3faff"
+
+# bazel build does not work with earlier versions:
+#
+# ERROR: /Users/jml/go/src/github.com/weaveworks/cortex/vendor/github.com/weaveworks-experiments/loki/pkg/client/BUILD.bazel:3:1: no such package 'vendor/github.com/openzipkin/zipkin-go-opentracing/_thrift/gen-go/zipkincore': BUILD file not found on package path and referenced by '//vendor/github.com/weaveworks-experiments/loki/pkg/client:go_default_library'.
```
Comment (tomwilkie) [inline Gopkg.toml]: Have you got an up to date version of gazelle?

Comment (jml) [inline Gopkg.toml]: Apparently not.

**Comment by jml** [comment]
Forgot to test last two commits with `make bazel bazel-test`, so have reverted them to go back to known good state.

**Comment by tomwilkie** [comment]
I'm +1 on explicit dependancies and +1 on the version upgrades, but -1 on the override.  Let me know when you get it working and I'll take another look.  When you want to build this in CI (and want to build the protos with bazel) give me a shout, as I have a branch that already does it somewhere.

**Comment by jml** [comment]
Thanks. I'll look into it. Why are you against the override?

**Comment by tomwilkie** [comment]
> Why are you against the override?

It should be unnecessary with the latest gazelle, unless I'm missing
something.

**Comment by jml** [comment]
I haven't forgotten this. I'm just on a tight deadline with some other project work.

> When you want to build this in CI (and want to build the protos with bazel) give me a shout, as I have a branch that already does it somewhere.

@tomwilkie Could you please point me at that? I would like to do that shortly after merging this.

**Comment by jml** [comment]
Closed in favour of #557

**Pull request closed without merging**